src/ServiceControl/HeartbeatMonitoring/HeartbeatsMonitor.cs

```diff
@@ -56,7 +58,7 @@ void Refresh(object _)
 
                 try
                 {
-                    timer.Change((int)TimeSpan.FromSeconds(5).TotalMilliseconds, -1);
+                    timer.Change(TimeSpan.FromSeconds(5), Timeout.InfiniteTimeSpan);
```

Comment:
PeriodicExecuter uses Tasks + Task.Delay to in essence replicate what a Timer gives us for free.
In other words, I am not sure we should be using it at all